src/internal/observable/innerFrom.ts

```diff
@@ -12,6 +12,7 @@ import { isFunction } from '../util/isFunction';
 import { reportUnhandledError } from '../util/reportUnhandledError';
 import { observable as Symbol_observable } from '../symbol/observable';
 
+export function innerFrom<O extends ObservableInput<any>>(input: O): Observable<ObservedValueOf<O>>;
```

Comment:
Can we just change the signature instead of adding an overload signature - which will hide the implementation signature anyway?

Reply:
I did that and the guts of the implementation hated me for it. I'll fiddle with it again before I merge it, but I figure it's not super important since it's not a publicly exposed function.

Reply:
Upon review. I'm going to do a follow up PR on this, because I think maybe there are some issues with `onErrorResumeNext`'s types.